app/gui/view/component-browser/component-list-panel/grid/src/entry/style.rs

```diff
@@ -97,75 +176,72 @@ pub struct Style {
-/// computed by mixing "main color" with application background.
-///
-/// `icon_strong` and `icon_weak` parameters represent the more/less contrasting parts of the
-/// [icon](crate::icon::Any), they do not represent highlighted state of the icon.
+/// This structure can be created from a single "main color" input. Each of these colors can be
+/// computed by modifying the transparency of the "main color". See [`Color`] for more information.
 #[allow(missing_docs)]
 #[derive(Clone, CloneRef, Debug)]
-pub struct Colors {
+pub struct ResolvedColors {
     pub background:      frp::Sampler<color::Lcha>,
     pub hover_highlight: frp::Sampler<color::Lcha>,
     pub text:            frp::Sampler<color::Lcha>,
-    pub icon_strong:     frp::Sampler<color::Lcha>,
-    pub icon_weak:       frp::Sampler<color::Lcha>,
+    pub icon:            frp::Sampler<color::Lcha>,
     pub skip_animations: frp::Any,
 }
 
-impl Colors {
+impl ResolvedColors {
     /// Create color outputs from given main color and style.
     ///
-    /// Each of these colors will be computed by mixing "main color" with application background
-    /// according to the proper field in [`Style::color_intensities`]. The dimming of group is
-    /// animated.
+    /// Each of these colors can be computed by modifying the transparency of the "main color". See
+    /// [`Color`] for more information. All color changes are animated.
     pub fn from_main_color(
         network: &frp::Network,
         style_watch: &StyleWatchFrp,
-        color: &frp::Stream<color::Lcha>,
-        color_intensities: &frp::Stream<ColorIntensities>,
+        main_color: &frp::Stream<color::Lcha>,
+        colors: &frp::Stream<Colors>,
         is_dimmed: &frp::Stream<bool>,
     ) -> Self {
-        fn mix((c1, c2): &(color::Lcha, color::Lcha), coefficient: &f32) -> color::Lcha {
-            color::mix(*c1, *c2, *coefficient)
-        }
-        let app_bg = style_watch.get_color(ensogl_hardcoded_theme::application::background);
-        let color_intensities = color_intensities.clone_ref();
+        let bg = style_watch.get_color(panel_theme::background_color);
+        let colors = colors.clone_ref();
+        let color_anim = color::Animation::new(network);
 
-        let intensity = Animation::new(network);
         frp::extend! { network
             init <- source_();
 
-            text_intensity <- color_intensities.map(|c| c.text);
-            bg_intensity <- color_intensities.map(|c| c.background);
-            hover_hg_intensity <- color_intensities.map(|c| c.hover_highlight);
-            dimmed_intensity <- color_intensities.map(|c| c.dimmed);
-            icon_strong_intensity <- color_intensities.map(|c| c.icon_strong);
-            icon_weak_intensity <- color_intensities.map(|c| c.icon_weak);
-
-            one <- init.constant(1.0);
+            bg_intensity <- colors.map(|c| c.background_intensity);
+            bg <- all_with(&bg, &init, |col, ()| color::Lcha::from(col));
```

Comment:
I got a bit confused here. We have `bg` which seems to be _panel_ background, and `bg_intensity` which in turns means _entry_ background. Please change those names to have clear distinction when we have panel background and when entry background.

Reply:
Changed names and reordered the lines slightly. This should be more readable now.